spec/requests/provider_interface/offers_controller_spec.rb

```diff
@@ -68,4 +68,63 @@
       end
     end
   end
+
+  # rubocop:disable RSpec/AnyInstance
+  describe 'validation errors' do
+    let(:application_choice) do
+      create(
+        :application_choice, :offer,
+        application_form: application_form,
+        course_option: course_option
+      )
+    end
+
+    before { allow_any_instance_of(ProviderInterface::OfferWizard).to receive(:valid_for_current_step?).and_return(false) }
+
+    it 'tracks errors on create' do
+      application_choice.update(status: 'awaiting_provider_decision')
+      expect {
+        post provider_interface_application_choice_offer_path(application_choice)
+      }.to change(ValidationError, :count).by(1)
+    end
+
+    it 'tracks errors on (providers) update' do
```

Comment:
We don't currently cover the create actions with these specs.

Could we commonize this sort of spec into a shared example so these files look more like a list of endpoints which should track validation errors?

Reply:
:+1: Thanks @benswannack , I've added missing create actions (my routes grepping let me down there). Not a huge fan of shared examples outside the spec itself as it adds indirection so I've amended to add a shared example in the offers controller spec common to the offers and sub contollers actions (ie not common across all similar controller specs).